lib/jar.ts

```diff
@@ -0,0 +1,22 @@
+import * as path from 'path';
+import * as crypto from 'crypto';
+import * as fs from 'fs';
+
+export function getTempPomPathForJar(root: string, jar: string): string {
+  const sha1 = getSha1ForJar(path.join(root, jar));
+
+  // return path to root of pom.xml
+  return '';
+}
+
+export function isJar(file: string) {
+  return file.match(/\.(([jw]ar)|(zip))$/);
+}
+
+export function getSha1ForJar(filePath) {
+  const fileContents = fs.readFileSync(filePath);
+  return crypto
+    .createHash('sha1')
+    .update(fileContents)
+    .digest('hex');
```

Comment:
make the function a bit more flexible, add the algorithm (sha1) and encoding (hex) as parameters to the function with defaults? we may need to tweak them based on the platform we run on:

from docs: "The algorithm is dependent on the available algorithms supported by the version of OpenSSL on the platform." - we will need to ensure that our maven-deps service has OpenSSL installed and the version supports creating sha1 hashes

Reply:
we are calling maven-central directly, which only supports getting packages details for sha1

Reply:
ok this could be a constant with a comment to that effect?

Reply:
with the CLI, does the user need OpenSSL installed?